### PR TITLE
Support `drop_last` Argument in cuGraph-PyG Loader

### LINKED ISSUE #3949: [IMP] Accept the shuffle and drop_last arguments in the cuGraph-PyG loader
Currently, passing these arguments results in an exception.  They should be handled properly, which could be returning a warning and not modifying the behavior of cuGraph.

### PR DESCRIPTION
**Pull request by alexbarghi-nv**
Supports the `drop_last` argument in cuGraph-PyG for better compatibility with native PyG workflows.

Closes #3949 

Merge after #3985

### CODE REVIEW

File: python/cugraph-pyg/cugraph_pyg/data/cugraph_store.py [hunk 1]
```diff
@@ -364,6 +368,24 @@ def __infer_offsets(
...
+        del split_array
+
+        dask_array = dar.concatenate(
+            [
+                dar.from_delayed(s, shape=shapes[i], dtype=array.dtype)
+                for i, s in enumerate(scattered_array)
+            ]
+        )
+        del scattered_array
+
```
Comment (VibhuJawa) [inline python/cugraph-pyg/cugraph_pyg/data/cugraph_store.py:386]: We should use `da.from_array(x)  ` here. 

https://docs.dask.org/en/stable/generated/dask.array.from_array.html

**Review by VibhuJawa — CHANGES_REQUESTED** [review]
Requested changes around dask, looks good otherwise.

File: python/cugraph-pyg/cugraph_pyg/data/cugraph_store.py (hunk 1 above)
Comment (alexbarghi-nv) [reply python/cugraph-pyg/cugraph_pyg/data/cugraph_store.py]: fixed

**Review by tingyu66 — CHANGES_REQUESTED** [review]
Please see my comments.

File: python/cugraph-pyg/cugraph_pyg/loader/cugraph_node_loader.py [hunk 2]
```diff
@@ -197,11 +211,14 @@ def __init__(
 
         # Truncate if we can't evenly divide the input array
         stop = (len(input_nodes) // batch_size) * batch_size
-        input_nodes = input_nodes[:stop]
+        input_nodes, remainder = cupy.array_split(input_nodes, [stop])
 
         # Split into batches
         input_nodes = cupy.split(input_nodes, len(input_nodes) // batch_size)
```
[the remark this answers is not in the document]
Comment (alexbarghi-nv) [reply python/cugraph-pyg/cugraph_pyg/loader/cugraph_node_loader.py]: I think it should be fixed now

[the remark this answers is not in the document]
Comment (tingyu66) [reply python/cugraph-pyg/cugraph_pyg/loader/cugraph_node_loader.py]: LGTM

**Review by tingyu66 — APPROVED** [review]
:+1:

**Review by BradReesWork — APPROVED** [review]

**Review by VibhuJawa — APPROVED** [review]
Looks good to merge once  https://github.com/rapidsai/cugraph/pull/3985  is  merged intio this. LGTM

**Pull request merged**